incydr/_core/models.py

```diff
@@ -102,3 +128,62 @@ def expired(self):
         return (datetime.utcnow() - self._init_time) > timedelta(
             seconds=self.expires_in
         )
+
+
+class CSVModel(BaseModel, allow_population_by_field_name=True, extra="allow"):
+    """
+    Pydantic model class enables multiple aliases to be assigned to a single field value. If the field is required
+    then at least one of the aliases must be supplied or validation will fail.
+
+    Useful when parsing CSV data from multiple sources where the expected column header names might vary.
+
+    For example, if a CSV requires a "user" column, which could either be a username or ID:
+
+        class UserCSV(CSVModel):
+            user: str = Field(csv_aliases=["user_id", "userId", "username"])
```

Comment:
is the field name also a valid alias in this case (ie. `user`) ?

Reply:
In this example, no. But if one wanted it to be an alias, they could just add it to the list.